### PR TITLE
test(color-picker): alpha support

### PR DESCRIPTION
**Pull request by alisonailea**
**Related Issue:** #749

## Summary

Cleans up tests for alpha feature on color picker. There is something wrong with Puppeteer causing an unknown error when puppeteer attempts to make programmatic changes to the alpha input box but these changes have been tested manually and are good to go for the 1.0 release. A ticket has been made to address the unstable tests.

### CODE REVIEW

File: src/components/color-picker/color-picker.tsx [hunk 1]
```diff
@@ -974,10 +974,7 @@ export class ColorPicker
   }
 
   private renderOpacitySection(): VNode {
-    const { color, previousColor } = this;
-    const sliderOpacity = alphaToOpacity((color ? color : previousColor).alpha()); // slider keeps previous alpha when null
-    const inputOpacity = color ? alphaToOpacity(color.alpha()).toString() : undefined;
```
Comment (benelan) [inline src/components/color-picker/color-picker.tsx]: Why was this changed to match the sliderOpacity value? I assume there was a reason they were different in the first place, although I have no idea what it might be lol

File: src/components/color-picker/color-picker.e2e.ts [hunk 2]
```diff
@@ -1060,61 +1142,124 @@ describe("calcite-color-picker", () => {
...
+            await rInput.callMethod("setFocus");
+            await selectText(rInput);
+            await page.keyboard.press("Delete");
+            await page.keyboard.press("Enter");
+            await page.waitForChanges();
 
-              await page.keyboard.press("ArrowUp");
-              await page.waitForChanges();
-              expect(await calciteInput.getProperty("value")).toBe(consistentRgbHsvChannelValue);
+            expect(await rInput.getProperty("value")).toBe("");
```
Comment (benelan) [inline src/components/color-picker/color-picker.e2e.ts:1163]: Why did this change from "0" to "", was that a result of changes to the test or the component code?

File: package.json [hunk 3]
```diff
@@ -138,6 +139,7 @@
     "prettier": "2.8.1",
     "puppeteer": "10.0.0",
     "quicktype-core": "6.1.0",
+    "react-textarea-autosize": "8.4.0",
```
Comment (benelan) [inline package.json:142]: Adding these dependencies was to fix the storybook type errors right? Can you move these to a separate PR? The way our tsconfig/eslint is set up storybook type errors shouldn't effect builds or lintstaged. I'm pretty sure I asked Franco about adding these awhile ago so I want his eyes on the separate PR

**Review by benelan — COMMENTED** [review]
Looking good pretty good, thanks for taking this on! Did you clean up some tests that weren't related to alpha in this commit?

https://github.com/Esri/calcite-components/pull/6265/commits/8a610f7dc614b745d186c7979fc756e87ade1469

If so could you move them to a separate PR please

File: package.json (hunk 3 above)
Comment (alisonailea) [reply package.json:142]: I don't know what this is. It was already a part of the code base.

File: src/components/color-picker/color-picker.e2e.ts (hunk 2 above)
Comment (alisonailea) [reply src/components/color-picker/color-picker.e2e.ts:1163]: changes to the component code. The test was always resulting in ""

File: src/components/color-picker/color-picker.tsx (hunk 1 above)
Comment (alisonailea) [reply src/components/color-picker/color-picker.tsx]: the only change here was to make the code more readable. using "this." rather than destructuring at the top of the page and there is no need for "inputOpacity" as this.color is a required prop with a fallback.

**Comment by alisonailea** [comment]
> Looking good pretty good, thanks for taking this on! Did you clean up some tests that weren't related to alpha in this commit?

I kept running into weird timeout errors in the tests which I suspect were caused by the layered nested functions in the multi-layered "describes" these were resolved when I made each test stand alone.
